backends/gpu/lib/kernels/dnn_kernels.cc

```diff
@@ -169,13 +170,35 @@ static Error DnnPoolingBackward(
   if (auto error = wrapper::DnnSetStream(handle.get(), stream.get()))
     return error;
 
+  auto platform = handle->platform();
+
   wrapper::Pointer<const void> alpha_ptr(&alpha, handle->platform());
   wrapper::Pointer<const void> beta_ptr(&beta, handle->platform());
 
-  return wrapper::DnnPoolingBackward(
-      *current, handle.get(), pooling_desc.get(), alpha_ptr, y_desc.get(),
-      y.pointer(), dy_desc.get(), dy.pointer(), x_desc.get(), x.pointer(),
-      beta_ptr, dx_desc.get(), dx.pointer());
+  switch (platform) {
+    case wrapper::Platform::CUDA:
+      return wrapper::CudnnPoolingBackward(
+          *current, handle.get(), pooling_desc.get(), alpha_ptr, y_desc.get(),
+          y.pointer(), dy_desc.get(), dy.pointer(), x_desc.get(), x.pointer(),
+          beta_ptr, dx_desc.get(), dx.pointer());
+    case wrapper::Platform::ROCm:
+    {
+      size_t workspace_size;
+      wrapper::Pointer<void> workspace_size_ptr(&workspace_size, platform);
+      if (MiopenPoolingGetWorkSpaceSizeV2(*current, pooling_desc.get(), y_desc.get(), workspace_size_ptr)){
+        return MakeStringError("Unable to get the workspace size.");
```

Comment:
Please return the error from `MiopenPoolingGetWorkSpaceSize()` instead of creating another.